### PR TITLE
test(utils): clean up existing tests, unify setup

### PR DESCRIPTION
**Pull request by Tlacenka**
This is the first PR regarding utils test refactor in #262 . In it, I introduce the following changes:
- I unify the mocks setup. Since the multi-progress-bars integration overrides console, I had to adapt the setup. All vitest configuration files now mock console by default.
- I removed global setup for `tmp` folder that should no longer be used in unit or integration tests
- I unify the colour range limit values used in multiple functions.
- I introduce a smaller `REPORT_MOCK` for testing purposes.
- I unify naming of transformation function file to `transform`.
- I introduce the `mocks` folder in the `utils` package.
- I clean up the tests and move `executeProcess` testing utilities to `testing-utils`.

### CODE REVIEW

**Review by MishaSeredenkoPushBased — DISMISSED** [review]

**Review by matejchalk — COMMENTED** [review]
Overall a nice refactor :slightly_smiling_face: Left some comments.

File: packages/utils/vite.config.unit.ts [hunk 1]
```diff
@@ -12,7 +12,6 @@ export default defineConfig({
     },
     environment: 'node',
     include: ['src/**/*.unit.test.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
-    globalSetup: ['global-setup.ts'],
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply packages/utils/vite.config.unit.ts:15]: The global setup is currently not in every vitest config file at the moment. I will do that instead then.

File: testing-utils/src/lib/setup/console.mock.ts [hunk 2]
```diff
@@ -5,13 +5,32 @@ let consoleWarnSpy: SpyInstance;
 let consoleErrorSpy: SpyInstance;
 
 beforeEach(() => {
-  consoleInfoSpy = vi.spyOn(console, 'info');
-  consoleWarnSpy = vi.spyOn(console, 'warn');
-  consoleErrorSpy = vi.spyOn(console, 'error');
+  // @ts-ignore In multi-progress-bars, console methods are overriden
+  if (console.info) {
+    consoleInfoSpy = vi.spyOn(console, 'info');
+  }
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply testing-utils/src/lib/setup/console.mock.ts:11]: Unfortunately, there was no guide that I could find about how to handle this gracefully. The typescript alert did not offer any solution. Will do.

File: packages/utils/src/lib/scoring.unit.test.ts [hunk 3]
```diff
@@ -1,81 +1,82 @@
...
+        { slug: 'cumulative-layout-shift', weight: 1, score: 1 },
+        { slug: 'speed-index', weight: 0, score: 1 },
+      ],
+      0.25,
+    ],
+  ])(
+    'should calculate score based on individual scores and weights',
+    (refs, score) => {
+      expect(
+        calculateScore(refs, (ref: (typeof refs)[number]) => ref.score),
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply packages/utils/src/lib/scoring.unit.test.ts:27]: Weird, it was shouting at me when I did not include the type.

File: packages/utils/src/lib/scoring.unit.test.ts [hunk 4]
```diff
@@ -1,81 +1,82 @@
...
+    [
+      [
+        { slug: 'first-contentful-paint', weight: 3, score: 0 },
+        { slug: 'cumulative-layout-shift', weight: 1, score: 1 },
+        { slug: 'speed-index', weight: 0, score: 1 },
+      ],
+      0.25,
+    ],
+  ])(
+    'should calculate score based on individual scores and weights',
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply packages/utils/src/lib/scoring.unit.test.ts:24]: Tried, wondering if in this case splitting it or having an object is better.

**Review by matejchalk — APPROVED** [review]

**Pull request merged**